File: frontend/edenBooks/src/modules/addresses/services/addressService.ts
import { addressApi } from "../api/addressApi";
import { IAddress } from "../interfaces/IAddress";
import { IAddressService } from "../interfaces/IAddressService";

export const AddressService: IAddressService = {
  CreateAddress: async (address: IAddress) =>addressApi.CreateAddress(address),
  GetAddresss: async () => addressApi.GetAddresss(),
  GetAddressById: async (id: number) => addressApi.GetAddressById(id),
  UpdateAddress: async (id: number, address: IAddress) =>addressApi.UpdateAddress(id, address),
  DeleteAddress: async (id: number) => addressApi.DeleteAddress(id),
};

File: backend/internal/modules/addresses/domain/ports/services/address_service.go
package services

import (
	"github.com/javierjpv/edenBooks/internal/modules/addresses/domain/entities"
)

type AddressService interface {
	CreateAddress(a *entities.Address) (*entities.Address, error)

	UpdateAddress(a *entities.Address) error

	DeleteAddress(id uint) error

	GetAddressByID(id uint) (*entities.Address, error)

	CheckExistingAdress(a entities.Address) (bool, error)

	GetFilteredAddresses(filters map[string]string) ([]entities.Address, error)
}
